### DISCUSSION TITLE
Auto-approve edits within the repository (not all files on computer)

### DISCUSSION THREAD

**Discussion by domdomegg** [cline/cline#991] [Feature Requests]
At the moment, the auto-approve edits permission appears to give the model permission to edit whatever files it wants on your computer ("Allows modification of any files on your computer.").

<img width="304" alt="image" src="https://github.com/user-attachments/assets/5e5f4ece-7626-45f1-bdc7-26628fee1fe9" />

This seems more broadly scoped than I would hope - this could be quite destructive and almost always I never want Cline to be editing outside the VS Code workspace.

I'd like to propose changing this either by default (or with an option) to being scoped to the VS Code workspace. This limits the blast radius of bad changes, and usually is protected by version control etc.

---

This might be already enforced, and this wording might just be confusing in the UI. If that is the case, then this is a feature request to change the wording to something like 'in your workspace' rather than 'on your computer'.

**Comment by gkiely** [comment]
I would love to see this.

**Comment by 625781186** [comment]
<img width="612" height="849" alt="image" src="https://github.com/user-attachments/assets/964dd8b1-3cfc-402d-a771-ffff7f0d18a7" />

Have this been implemented yet?

kilocode/cursor and other can add custom command to auto-approve but Cline can't .

Discussion https://github.com/cline/cline/discussions/1523: Highlighted the need for regex-based command restrictions.
Discussion https://github.com/cline/cline/discussions/1745: Proposed allowing users to bypass approval for specific commands.
Discussion  https://github.com/cline/cline/discussions/2253: Command Auto-Approval Rules
